features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/domain/ResourceRequestFactoryTest.kt

```diff
@@ -0,0 +1,166 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.domain
+
+import com.datadog.android.DatadogSite
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.context.DatadogContext
+import com.datadog.android.api.feature.Feature
+import com.datadog.android.api.net.Request
+import com.datadog.android.api.net.RequestFactory
+import com.datadog.android.api.net.RequestFactory.Companion.HEADER_API_KEY
+import com.datadog.android.api.net.RequestFactory.Companion.HEADER_EVP_ORIGIN
+import com.datadog.android.api.net.RequestFactory.Companion.HEADER_EVP_ORIGIN_VERSION
+import com.datadog.android.api.storage.RawBatchEvent
+import com.datadog.android.sessionreplay.forge.ForgeConfigurator
+import com.datadog.android.sessionreplay.internal.domain.ResourceRequestFactory.Companion.APPLICATION_ID
+import com.datadog.android.sessionreplay.internal.domain.ResourceRequestFactory.Companion.UPLOAD_DESCRIPTION
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import okhttp3.MediaType
+import okhttp3.MultipartBody
+import okhttp3.RequestBody
+import okio.Buffer
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(ForgeConfigurator::class)
+internal class ResourceRequestFactoryTest {
+    private lateinit var testedRequestFactory: ResourceRequestFactory
+
+    @Mock
+    lateinit var mockResourceRequestBodyFactory: ResourceRequestBodyFactory
+
+    private lateinit var fakeRawBatchEvents: List<RawBatchEvent>
+
+    @Mock
+    lateinit var mockRequestBody: RequestBody
+
+    @Mock
+    lateinit var mockInternalLogger: InternalLogger
+
+    @Forgery
+    lateinit var fakeRawBatchEvent: RawBatchEvent
+
+    private lateinit var fakeMediaType: MediaType
+
+    @Mock
+    lateinit var mockDatadogSite: DatadogSite
+
+    @StringForgery
+    lateinit var fakeApplicationId: String
+
+    @Mock
+    lateinit var fakeDatadogContext: DatadogContext
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        val fakeRumFeature = HashMap<String, String>()
```

Comment:
minor suggestion. The same for other similar instances

```suggestion
        val fakeRumFeature = mapOf<String, String>(APPLICATION_ID to fakeApplicationId)
```